### PR TITLE
Added doctests to bucket sort

### PR DESCRIPTION
**Pull request by mateuszz0000**
### **Describe your change:**
Added missing doctests.

* [ ] Add an algorithm?
* [x] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: sorts/bucket_sort.py [hunk 1]
```diff
@@ -1,33 +1,51 @@
...
+algorithm, or by recursively applying the bucket sorting algorithm. It is a
+distribution sort, and is a cousin of radix sort in the most to least
+significant digit flavour.
+Bucket sort is a generalization of pigeonhole sort. Bucket sort can be
+implemented with comparisons and therefore can also be considered a
+comparison sort algorithm. The computational complexity estimates involve the
+number of buckets.
+
+Time Complexity of Solution:
+Worst case scenario occurs when all the elements are placed in a single bucket. The overall performance
```
Comment (cclauss) [inline sorts/bucket_sort.py:21]: Please wrap lines to 88 characters max.

**Comment by Bhavik-Jikadara** [comment]
What is it

**Comment by cclauss** [comment]
Is it a problem that the algorithm uses the builtin `sorted()` function?

**Comment by mateuszz0000** [comment]
@cclauss should we merge it?

File: sorts/bucket_sort.py [hunk 2]
```diff
@@ -1,5 +1,3 @@
-#!/usr/bin/env python3
```
Comment (cclauss) [inline sorts/bucket_sort.py]: Why remove the shebang line?  https://en.wikipedia.org/wiki/Shebang_(Unix)

Comment (mateuszz0000) [reply sorts/bucket_sort.py]: My bad.

**Pull request merged**